### PR TITLE
One demo.py  file for saving the results of multi-images after detecting images.

### PR DESCRIPTION
**Pull request by AndyVerne**
## Motivation
To view the results of  conveniently. Hope this project can save the images after processing multi-images(*.PNG format). So I made some codes and changes based on this project.

## Modification
1. Made one new file whose demo.py in /tools folder to save images after detecting. 
2. According to former point, made some changes to mmdet/apis/inference.py and mmdet/models/detectors/rpn.py.

## Use cases 
python tools/demo.py configs/config-file checkpoints/checkpoint-file demo/folder-store-original_images result_demo/folder-store-detected_images
eg.
python tools/demo.py configs/faster_rcnn/faster_rcnn_r50_fpn_1x_coco.py checkpoints/faster_rcnn_r50_fpn_1x_coco_20200130-047c8118.pth demo/ result_demo/

### CODE REVIEW

**Comment by hhaAndroid** [comment]
> ## Motivation
> To view the results of conveniently. Hope this project can save the images after processing multi-images(*.PNG format). So I made some codes and changes based on this project.
> 
> ## Modification
> 1. Made one new file whose demo.py in /tools folder to save images after detecting.
> 2. According to former point, made some changes to mmdet/apis/inference.py and mmdet/models/detectors/rpn.py.
> 
> ## Use cases
> python tools/demo.py configs/config-file checkpoints/checkpoint-file demo/folder-store-original_images result_demo/folder-store-detected_images
> eg.
> python tools/demo.py configs/faster_rcnn/faster_rcnn_r50_fpn_1x_coco.py checkpoints/faster_rcnn_r50_fpn_1x_coco_20200130-047c8118.pth demo/ result_demo/

Thank you for your contribution！If you want to achieve the above features, I suggest it be compatible with demo/demo.py.

**Comment by AndyVerne** [comment]
> > ## Motivation
> > To view the results of conveniently. Hope this project can save the images after processing multi-images(*.PNG format). So I made some codes and changes based on this project.
> > ## Modification
> > 
> > 1. Made one new file whose demo.py in /tools folder to save images after detecting.
> > 2. According to former point, made some changes to mmdet/apis/inference.py and mmdet/models/detectors/rpn.py.
> > 
> > ## Use cases
> > python tools/demo.py configs/config-file checkpoints/checkpoint-file demo/folder-store-original_images result_demo/folder-store-detected_images
> > eg.
> > python tools/demo.py configs/faster_rcnn/faster_rcnn_r50_fpn_1x_coco.py checkpoints/faster_rcnn_r50_fpn_1x_coco_20200130-047c8118.pth demo/ result_demo/
> 
> Thank you for your contribution！If you want to achieve the above features, I suggest it be compatible with demo/demo.py.

Thanks for your advice! I moved the demo.py under /tools folder to the /demo folder. It can be compatible with demo/demo.py.

**Comment by hhaAndroid** [comment]
> > > ## Motivation
> > > To view the results of conveniently. Hope this project can save the images after processing multi-images(*.PNG format). So I made some codes and changes based on this project.
> > > ## Modification
> > > 
> > > 1. Made one new file whose demo.py in /tools folder to save images after detecting.
> > > 2. According to former point, made some changes to mmdet/apis/inference.py and mmdet/models/detectors/rpn.py.
> > > 
> > > ## Use cases
> > > python tools/demo.py configs/config-file checkpoints/checkpoint-file demo/folder-store-original_images result_demo/folder-store-detected_images
> > > eg.
> > > python tools/demo.py configs/faster_rcnn/faster_rcnn_r50_fpn_1x_coco.py checkpoints/faster_rcnn_r50_fpn_1x_coco_20200130-047c8118.pth demo/ result_demo/
> > 
> > 
> > Thank you for your contribution！If you want to achieve the above features, I suggest it be compatible with demo/demo.py.
> 
> Thanks for your advice! I moved the demo.py under /tools folder to the /demo folder. It can be compatible with demo/demo.py.

I think it can be directly compatible with demo/demo.py, without the need to rewrite tools/demo.py.

File: mmdet/models/detectors/rpn.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import mmcv
 import torch
-from mmcv.image import tensor2imgs
+from mmcv.image import tensor2imgs,imwrite
```
Comment (ZwwWayne) [inline mmdet/models/detectors/rpn.py:3]: imwtite seems to be unnecessary here.

Comment (AndyVerne) [reply mmdet/models/detectors/rpn.py]: Thanks for your reminding. I deleted it.

**Comment by hhaAndroid** [comment]
Hi @AndyVerne Can you make the above functions compatible with demo/image_demo.py？

**Comment by hhaAndroid** [comment]
@AndyVerne It seems that we can do it through `image_demo.py`, so we temporarily close this PR. If there are new developments in the follow-up, you can open it.

**Comment by OpenMMLab-Coodinator** [comment]
Hi @AndyVerne  ！We are grateful for your efforts in helping improve this open-source project during your personal time. 
Welcome to join OpenMMLab Special Interest Group (SIG) , where you can share your experiences, ideas, and build connections with like-minded peers. 

To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/UjgXkPWNqA

If you have a WeChat account，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution❤

**Pull request closed without merging**